Explorer/Assets/Scripts/CrdtEcsBridge/WorldSynchronizer/CommandBufferSynchronizer/SDKComponentCommandBufferSynchronizer.cs

```diff
@@ -34,14 +35,18 @@ public override void Apply(World world, PersistentCommandBuffer commandBuffer, E
             {
                 case CRDTReconciliationEffect.ComponentModified:
                     // if component is modified then return to the pool the existing one
-                    componentPool.Release(world.Get<T>(entity));
-                    commandBuffer.Set(entity, c);
+                    // No need to add it to the command buffer as we already get the component by ref and can "override" it directly without overhead
+                    ref T pointerToPrevObj = ref world.Get<T>(entity);
+                    componentPool.Release(pointerToPrevObj);
+                    pointerToPrevObj = c;
```

Comment:
I assume `Debug.Assert(!world.Has<T>(entity)); ` can decrease the performance @m3taphysics

Reply:
As we spoke with @m3taphysics my changes made functional other scenes with very heavy logic with Character/Camera Area. There is already a ticket to optimize them.
Potentially `Debug.Assert` introduces overhead but it will be not present in the build